diff --git a/src/ext.rs b/src/ext.rs
index c34ffdc..5b5f52f 100644
--- a/src/ext.rs
+++ b/src/ext.rs
@@ -13,6 +13,8 @@ impl Protocol {
     pub const WEB_TRANSPORT: Protocol = Protocol(ProtocolInner::WebTransport);
     /// RFC 9298 protocol
     pub const CONNECT_UDP: Protocol = Protocol(ProtocolInner::ConnectUdp);
+    /// RFC 9220 WebSocket over HTTP/3 protocol
+    pub const WEBSOCKET: Protocol = Protocol(ProtocolInner::WebSocket);
 
     /// Return a &str representation of the `:protocol` pseudo-header value
     #[inline]
@@ -20,6 +22,7 @@ impl Protocol {
         match self.0 {
             ProtocolInner::WebTransport => "webtransport",
             ProtocolInner::ConnectUdp => "connect-udp",
+            ProtocolInner::WebSocket => "websocket",
         }
     }
 }
@@ -28,6 +31,7 @@ impl Protocol {
 enum ProtocolInner {
     WebTransport,
     ConnectUdp,
+    WebSocket,
 }
 
 /// Error when parsing the protocol
@@ -40,6 +44,7 @@ impl FromStr for Protocol {
         match s {
             "webtransport" => Ok(Self(ProtocolInner::WebTransport)),
             "connect-udp" => Ok(Self(ProtocolInner::ConnectUdp)),
+            "websocket" => Ok(Self(ProtocolInner::WebSocket)),
             _ => Err(InvalidProtocol),
         }
     }
diff --git a/src/lib.rs b/src/lib.rs
index 7fce04e..d9d4765 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,6 +1,7 @@
 //! HTTP/3 client and server
 #![deny(missing_docs, clippy::self_named_module_files)]
 #![allow(clippy::derive_partial_eq_without_eq)]
+#![allow(dead_code, unused_imports, unused_variables, mismatched_lifetime_syntaxes)]
 
 pub mod client;
 
