--- .cargo/registry/src/index.crates.io-1949cf8c6b5b557f/h3-0.0.8/src/ext.rs	2006-07-24 05:21:28.000000000 +0400
+++ vendor/h3/src/ext.rs	2026-03-11 21:25:15.857980520 +0300
@@ -13,6 +13,8 @@
     pub const WEB_TRANSPORT: Protocol = Protocol(ProtocolInner::WebTransport);
     /// RFC 9298 protocol
     pub const CONNECT_UDP: Protocol = Protocol(ProtocolInner::ConnectUdp);
+    /// RFC 9220 WebSocket over HTTP/3 protocol
+    pub const WEBSOCKET: Protocol = Protocol(ProtocolInner::WebSocket);
 
     /// Return a &str representation of the `:protocol` pseudo-header value
     #[inline]
@@ -20,6 +22,7 @@
         match self.0 {
             ProtocolInner::WebTransport => "webtransport",
             ProtocolInner::ConnectUdp => "connect-udp",
+            ProtocolInner::WebSocket => "websocket",
         }
     }
 }
@@ -28,6 +31,7 @@
 enum ProtocolInner {
     WebTransport,
     ConnectUdp,
+    WebSocket,
 }
 
 /// Error when parsing the protocol
@@ -40,6 +44,7 @@
         match s {
             "webtransport" => Ok(Self(ProtocolInner::WebTransport)),
             "connect-udp" => Ok(Self(ProtocolInner::ConnectUdp)),
+            "websocket" => Ok(Self(ProtocolInner::WebSocket)),
             _ => Err(InvalidProtocol),
         }
     }
--- .cargo/registry/src/index.crates.io-1949cf8c6b5b557f/h3-0.0.8/src/lib.rs	2006-07-24 05:21:28.000000000 +0400
+++ vendor/h3/src/lib.rs	2026-03-11 21:41:44.999807941 +0300
@@ -1,6 +1,7 @@
 //! HTTP/3 client and server
 #![deny(missing_docs, clippy::self_named_module_files)]
 #![allow(clippy::derive_partial_eq_without_eq)]
+#![allow(dead_code, unused_imports, unused_variables, mismatched_lifetime_syntaxes)]
 
 pub mod client;
 
